app/client/ui/FormPage.ts

```diff
@@ -95,16 +95,16 @@ export class FormPage extends Disposable {
   }
 
   private async _handleFormSubmit(formElement: HTMLFormElement) {
-    await this._model.submitForm(new TypedFormData(formElement));
+    return await this._model.submitForm(new TypedFormData(formElement));
   }
 
-  private async _handleFormSubmitSuccess() {
+  private async _handleFormSubmitSuccess(recordId: number) {
     const formLayout = this._model.formLayout.get();
     if (!formLayout) { throw new Error('formLayout is not defined'); }
 
     const {successURL} = formLayout;
     if (successURL) {
-      const url = sanitizeHttpUrl(successURL);
+      const url = sanitizeHttpUrl(successURL)?.replace("{{ID}}", `${recordId}`);
```

Comment:
I'd suggest making an explicit choice of what to do when `recordId` is invalid (undefined or 0, or more simply `!recordId`). Perhaps keep as `{{ID}}` or replace always with `0`? Or maybe it makes more sense to just skip redirect in this case?